### PR TITLE
[msal-extensions] Change sample location. Add documentation and diagrams. 

### PR DESCRIPTION
**Pull request by sangonzal**

### CODE REVIEW

**Review by hectormmg — APPROVED** [review]
LGTM!

File: extensions/docs/msal-node-extensions.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import { FilePersistenceWithDataProtection, DataProtectionScope } from "@azure/msal-node-extensions";
+
+const cachePath = "path/to/cache/file.json"; 
+const dataProtectionScope = DataProtectionScope.CurrentUser;
+const optinalEntropy = "";
+const windowsPersistence = FilePersistenceWithDataProtection.create(cachePath, dataProtectionScope, optionalEntropy);
+```
+
+- cachePath is the path in the file system where the encrypted cache file will be stored. 
+- dataProtectionScope specifies the scope fo the data protection - either the current user or the local machine. You do not need a key to protect or unprotect the data. If you set the scope to CurrentUser, only applications running on your credentials can unprotect the data; howerver, that mean that any application running on your credentials can access the protected data. If you set the scope to LocalMachine, any full-trust application on the computer can unprotect, access, and modify the data.
```
Comment (jo-arroyo) [inline extensions/docs/msal-node-extensions.md:28]:
```suggestion
- dataProtectionScope specifies the scope of the data protection - either the current user or the local machine. You do not need a key to protect or unprotect the data. If you set the scope to CurrentUser, only applications running on your credentials can unprotect the data; however, that means that any application running on your credentials can access the protected data. If you set the scope to LocalMachine, any full-trust application on the computer can unprotect, access, and modify the data.
```

File: extensions/docs/msal-node-extensions.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+- optionalEntoropy specifies password or other additional entropy used to encrypt the data.
```
Comment (jo-arroyo) [inline extensions/docs/msal-node-extensions.md:29]:
```suggestion
- optionalEntropy specifies password or other additional entropy used to encrypt the data.
```

File: extensions/docs/msal-node-extensions.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+The FilePersistenceWithDataProtection use the Win32 CryptProtectData and CryptUnprotectData APIs. For more information on dataProtectionScope, or optionalEntropy, reference the documentation for those APIs.
```
Comment (jo-arroyo) [inline extensions/docs/msal-node-extensions.md:31]:
```suggestion
The FilePersistenceWithDataProtection uses the Win32 CryptProtectData and CryptUnprotectData APIs. For more information on dataProtectionScope, or optionalEntropy, reference the documentation for those APIs.  
```

File: extensions/samples/msal-node-extensions/index.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const express = require("express");
+const msal = require('@azure/msal-node');
```
Comment (jo-arroyo) [inline extensions/samples/msal-node-extensions/index.js:7]:
```suggestion
const msal = require("@azure/msal-node");
```

File: extensions/samples/msal-node-extensions/index.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    if (process.platform === "darwin") {
+        return extensions.KeychainPersistence.create(cachePath, "serviceName", "accountName"); // Replace serviceName and accountName
+    }
+
+    // On Linux, uses  libsecret to store to secret service. Libsecret has to be installed.
+    if (process.platform === "linux") {
+        return extensions.LibSecretPersistence.create(cachePath, "serviceName", "accountName"); // Replace serviceName and accountName
+    }
+
+    throw new Error("Could not create perrsistence. Platform not supported");
```
Comment (jo-arroyo) [inline extensions/samples/msal-node-extensions/index.js:88]:
```suggestion
    throw new Error("Could not create persistence. Platform not supported");
```

**Review by jo-arroyo — APPROVED** [review]
LGTM. Just a few nits

**Comment by sangonzal** [comment]
@jo-arroyo Thanks for the detailed review! I need to do a better job with spell checking documentation and error messages. 😅

**Review by sameerag — APPROVED** [review]

**Pull request merged**